modules/vislab/rendering/renderabledistancelabel.cpp

```diff
@@ -0,0 +1,107 @@
+ * OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.                                         *
+ ****************************************************************************************/
+
+#include <modules/vislab/rendering/renderabledistancelabel.h>
+
+#include <modules/base/rendering/renderablenodeline.h>
+#include <openspace/documentation/documentation.h>
+#include <openspace/documentation/verifier.h>
+#include <openspace/engine/globals.h>
+#include <openspace/rendering/renderengine.h>
+#include <openspace/scene/scene.h>
+#include <string>
+
+namespace {
+    constexpr const char* _loggerCat = "RenderableDistanceLabel";
+
+    constexpr openspace::properties::Property::PropertyInfo NodeLineInfo = {
+        "NodeLine",
+        "Node Line",
+        "Property to track a nodeline. When tracking the label text will be updating the distance "
+        "from the nodeline start and end. "
+    };
+}
+
+namespace openspace {
+
+documentation::Documentation RenderableDistanceLabel::Documentation() {
+    using namespace documentation;
+    return {
+        "Renderable Distance Label",
+        "vislab_renderable_distance_label",
+        {
+            {
+                NodeLineInfo.identifier,
+                new StringVerifier,
+                Optional::No,
+                NodeLineInfo.description
+            },
+        }
+    };
+}
+
+RenderableDistanceLabel::RenderableDistanceLabel(const ghoul::Dictionary& dictionary)
+    : RenderableLabels(dictionary)
+    , _nodeLine(NodeLineInfo)
+{
+    documentation::testSpecificationAndThrow(
+        Documentation(),
+        dictionary,
+        "RenderableDistanceLabel"
+    );
+
+    if (dictionary.hasKey(NodeLineInfo.identifier)) {
+        _nodeLine = dictionary.value<std::string>(NodeLineInfo.identifier);
+        addProperty(_nodeLine);
+    }
+}
+
+void RenderableDistanceLabel::update(const UpdateData& data) {
+
+    if (global::renderEngine.scene()->sceneGraphNode(_nodeLine)) {
+
+        // Calculate distance
+        SceneGraphNode* nodelineNode = global::renderEngine.scene()->sceneGraphNode(_nodeLine);
+        RenderableNodeLine* nodeline = dynamic_cast<RenderableNodeLine*>(nodelineNode->renderable());
+        double myDistance = nodeline->getDistance();
+
+        // format string
+        float scale = getUnit(Kilometer);
+        std::string distanceText = std::to_string(std::round(myDistance / scale));
+        int pos = distanceText.find(".");
+        std::string subStr = distanceText.substr(pos);
+        distanceText.erase(pos, subStr.size());
+        std::string finalText = distanceText + " Km";
+        setLabelText(finalText);
+
+        // Update placement of label with transformation matrix
+        glm::dvec3 start = global::renderEngine.scene()->sceneGraphNode(nodeline->_start)->worldPosition();
+        glm::dvec3 end = global::renderEngine.scene()->sceneGraphNode(nodeline->_end)->worldPosition();
```

Comment:
Is nodeline->_start/_end a guaranteed scenegraphnode if _nodeLine exists?

Reply:
They are only validated on initialization by the nodeline and are always of type scenegraphnode. However, we could mess up the validity on runtime since we are changing the id with properties from the gui. But IF those ID:s are wrong this will become very apparent as they will mess up the nodeline itself. If a check should be made I think this should be validated in renderablenodeline itself, not on the label.